lib/stream.js

```diff
@@ -80,27 +80,12 @@ const resumeStream = async stream => {
 
 const applyEncoding = (contents, encoding) => encoding === 'buffer' ? new Uint8Array(contents) : contents;
 
-// Retrieve streams created by the `std*` options
-const getCustomStreams = stdioStreamsGroups => stdioStreamsGroups.flat().filter(({type}) => type !== 'native');
-
-// Some `stdout`/`stderr` options create a stream, e.g. when passing a file path.
-// The `.pipe()` method automatically ends that stream when `childProcess.stdout|stderr` ends.
-// This makes sure we want for the completion of those streams, in order to catch any error.
-// Since we want to end those streams, they cannot be infinite, except for `process.stdout|stderr`.
-// However, for the `stdin`/`input`/`inputFile` options, we only wait for errors, not completion.
-// This is because source streams completion should end destination streams, but not the other way around.
-// This allows for source streams to pipe to multiple destinations.
-// We make an exception for `fileUrl` and `filePath`, since we create that source stream and we know it is piped to a single destination.
-const waitForCustomStreamsEnd = customStreams => customStreams
-	.filter(({value, type, direction}) => shouldWaitForCustomStream(value, type, direction))
-	.map(({value}) => finished(value));
-
-const throwOnCustomStreamsError = customStreams => customStreams
-	.filter(({value, type, direction}) => !shouldWaitForCustomStream(value, type, direction))
-	.map(({value}) => throwOnStreamError(value));
-
-const shouldWaitForCustomStream = (value, type, direction) => (type === 'fileUrl' || type === 'filePath')
-	|| (direction === 'output' && !isStandardStream(value));
+// Some `stdin`/`stdout`/`stderr` options create a stream, e.g. when passing a file path.
+// The `.pipe()` method automatically ends that stream when `childProcess` ends.
+// This makes sure we wait for the completion of those streams, in order to catch any error.
+const waitForCustomStreamsEnd = stdioStreamsGroups => stdioStreamsGroups.flat()
+	.filter(({type, value}) => type !== 'native' && !isStandardStream(value))
+	.map(({value}) => finished(value, {cleanup: true}));
```

Comment:
We now properly clean up all streams, whether input or output. In particular, if the child process errors, the input streams (such as a transform or a `ReadableStream` passed to the `stdin` option) are aborted.

So we can safely wait for their completion.